### PR TITLE
List API for slurm scheduler 

### PR DESCRIPTION
**Pull request by priyaramani**
Add API to list jobs for slurm scheduler
Address #503 

Test plan:
Unit test
```
[ec2-user@ip~]$ torchx list -s slurm
slurm://torchx/6
slurm://torchx/7
slurm://torchx/8
slurm://torchx/9
slurm://torchx/10
slurm://torchx/11
.
.
```

### CODE REVIEW

File: torchx/schedulers/slurm_scheduler.py [hunk 1]
```diff
@@ -557,7 +557,17 @@ def log_iter(
...
-        raise NotImplementedError()
+        # By default sacct only returns accounting informations of jobs launched on the current day
+        # To return all jobs launched, set starttime to one second past unix epoch time
+        # Starttime will be modified when listing jobs by timeframe is supported
+        p = subprocess.run(
+            ["sacct", "--json", "-S1970-01-01-00:00:01"],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        output_jsons = p.stdout.decode("utf-8")
```
Comment (kurman) [inline torchx/schedulers/slurm_scheduler.py:568]: nit: s/output_jsons/json_str?

File: torchx/schedulers/slurm_scheduler.py [hunk 2]
```diff
@@ -557,7 +557,17 @@ def log_iter(
         )
 
     def list(self) -> List[str]:
-        raise NotImplementedError()
+        # By default sacct only returns accounting informations of jobs launched on the current day
+        # To return all jobs launched, set starttime to one second past unix epoch time
+        # Starttime will be modified when listing jobs by timeframe is supported
+        p = subprocess.run(
+            ["sacct", "--json", "-S1970-01-01-00:00:01"],
```
Comment (kurman) [inline torchx/schedulers/slurm_scheduler.py:564]: Can this cause a problem if there are large number of jobs? Is there a notion of pagination?

File: torchx/schedulers/test/slurm_scheduler_test.py [hunk 3]
```diff
@@ -395,6 +395,19 @@ def test_describe_running(self, run: MagicMock) -> None:
         self.assertEqual(out.msg, "RUNNING")
         self.assertEqual(out.state, specs.AppState.RUNNING)
 
+    @patch("subprocess.run")
+    def test_list(self, run: MagicMock) -> None:
```
Comment (kurman) [inline torchx/schedulers/test/slurm_scheduler_test.py:399]: What do you think of adding integration tests as well?

Comment (priyaramani) [reply torchx/schedulers/test/slurm_scheduler_test.py:399]: Added

**Review by kurman — APPROVED** [review]

**Pull request closed without merging**